modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Editor.es.js

```diff
@@ -0,0 +1,833 @@
+							.querySelector('.option-selected');
+
+						await act(async () => {
+							fireEvent.click(operator);
+						});
+
+						await act(async () => {
+							fireEvent.click(getByText('Is equal to'));
+						});
+
+						const actionsType = document
+							.querySelectorAll('.timeline-item')[1]
+							.querySelectorAll('.ddm-field')[2]
+							.querySelector('.option-selected');
+
+						fireEvent.click(actionsType);
+
+						fireEvent.click(getByText('value'));
+
+						await waitForElement(() => {
+							return document.querySelector('.ddm-field');
+						});
+
+						if (firstArg === 'text') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('input.ddm-field-text')
+							).toBeTruthy();
+						}
+						else if (
+							firstArg === 'select' ||
+							firstArg == 'checkbox_multiple' ||
+							firstArg == 'radio'
+						) {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.form-builder-select-field')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'date') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.date-picker')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'color') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.clay-color-picker')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'image') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.input').disabled
+							).toBeTruthy();
+						}
+						else if (
+							firstArg === 'integer' ||
+							firstArg === 'double'
+						) {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('input.form-control')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'grid') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('table')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'rich_text') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.cke')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'document_library') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector(
+										'.liferay-ddm-form-field-document-library'
+									)
+							).toBeTruthy();
+						}
```

Comment:
@alinedoleron you can decrease this amount of conditions by grouping the selectors by type or by the test so you don't need a conditional, for example:

The `firstArg`, instead of just being the `value` of the fields, can be an object with `{type, selector}`, the selector being the respective field selector that helps you identify the element in the DOM. Once this was done, you would be able to remove all conditionals.

```js
expect(
	document
		.querySelectorAll('.timeline-item')[1]
		.querySelectorAll('.ddm-field')[3]
		.querySelector(selector)
).toBeTruthy();
```